lib/evmone/instructions.hpp

```diff
@@ -563,6 +563,23 @@ inline Result extcodecopy(StackTop stack, int64_t gas_left, ExecutionState& stat
     return {EVMC_SUCCESS, gas_left};
 }
 
+inline Result returndataload(StackTop stack, int64_t gas_left, ExecutionState& state) noexcept
+{
+    auto& index = stack.top();
+
+    const auto begin = static_cast<size_t>(index);
```

Comment:
Need to check before that `index < retudn_data.size()`, see `dataload` and `calldataload` implementations.

Otherwise this cast just ignores if high bits are set in 256-bit word.